docs/api.md

```diff
@@ -558,6 +559,57 @@ Response:
 }
 ```
 
+### `GET /api/saved_pages`
+
+Returns all a user's saved pages
+
+#### Request headers
+
+- `GOVUK-Account-Session`
+  - the user's session identifier
+
+#### JSON response fields
+
+- `govuk_account_session` *(optional)*
+  - a new session identifier
+- `saved_pages`
+  - an array of pages the user has saved, identified by their page path
+
+#### Response codes
+
+- 401 if the session identifier is invalid
+- 200 otherwise
+
+#### Example request / response
+
+Request (with gds-api-adapters):
+
+```ruby
+GdsApi.saved_page_api.get_saved_pages(
```

Comment:
if the `get_`, `set_`, `check_` isn't a convention then this could just be `.saved_pages`.

But perhaps we can see what names come out of the review of GDS API adaptors and I can check the docs match the reality once those are merged?

Reply:
I think the closest similar thing is the publishing-api's put content endpoint: https://github.com/alphagov/publishing-api/blob/main/docs/api.md#put-v2contentcontent_id

Which uses a PUT rather than a POST and has the ID as a path param, which makes sense as the ID is known without needing to talk to publishing-api first.  So I'd be in favour of making our save-a-new-page endpoint like that.

Reply:
For getting all content, the publishing-api uses [get_content_items](https://github.com/alphagov/gds-api-adapters/blob/master/lib/gds_api/publishing_api.rb#L352), so I think `get_saved_pages` is fine.